src/api/useAuth.tsx

```diff
@@ -54,7 +75,23 @@ export const useAuth = () => {
 			setUser(user);
 			addUserToDatabase(user);
 		});
+
+		// Cleanup the subscription when the component unmounts
+		return () => unsubscribe();
 	}, []);
 
 	return { user };
 };
+
+export const useGetUser = () => {
+	const { user } = useFindUser();
+	const navigate = useNavigate();
+
+	useEffect(() => {
+		if (user === null) {
+			navigate("/", { replace: true });
+		}
+	}, [user, navigate]);
```

Comment:
So right now this doens't actually only return the user it is just a differnet version of the `useFindUser` method. I have tried a few things but when I throw an error the whole app will crash just always hitting that error. 

I was trying to just put it in the `protectedRoute` and get the full change to work but I am a bit stuck. 😅 I have seemed to notic right now when I did a test console.log of it in `protectedRoute` that there seems to be 3 renders happening. I put it in a `useEffect` and did the console.log and still got those 3 console.logs with 2 being null and 1 being the data we need. So I do notice that with that it will automatically redirect because the first time it is `null`. It may be a needing to manage state in `useGetUser` like in `useFindUser` but I haven't quite determined that or what direction should it go.

Any suggestions?